### PR TITLE
feat: new cli for batch jobs

### PR DESCRIPTION
**Pull request by ivanleomk**
This targets #604 by implementing two main new features

1. A new `BatchJob` class which allows users to create `.jsonl` files which are compatible with OpenAI's new batch job function
2. a batch cli command that allows users to see all outstanding jobs, create a new batch job and cancel an existing batch jobs

----

| :rocket: | This description was created by [Ellipsis](https://www.ellipsis.dev) for commit db0ba2880f8a6f1eff96ff27cc27dca75e7dd191  | 
|--------|--------|

### Summary:
This PR introduces a new `BatchJob` class and CLI commands for managing batch jobs with OpenAI's API.

**Key points**:
- Added `BatchJob` class in `instructor/batch.py` for creating `.jsonl` files for batch jobs.
- Added CLI commands in `instructor/cli/batch.py` to list, create, and cancel batch jobs.
- Integrated new batch job commands into the main CLI in `instructor/cli/cli.py`.

----
Generated with :heart: by [ellipsis.dev](https://www.ellipsis.dev)

### CODE REVIEW

**Comment by jxnl** [comment]
please add a lil recording and doc pages before we're ready to merge

File: instructor/batch.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    method: Literal["POST"]
+    url: Literal["/v1/chat/completions"]
+    body: RequestBody
+
+
+class BatchJob:
+    @classmethod
+    def create_from_messages(
+        cls,
+        messages_batch: list[list[dict[str, Any]]],
```
Comment (jxnl) [inline instructor/batch.py:55]: just change this to list[Message]

**Review by jxnl — APPROVED** [review]

**Comment by ivanleomk** [comment]
This is a video recording of the cli in action where we use a generated `output.jsonl` file in order to create a batch job, cancel the same batch job and also list all existing batch jobs.

https://github.com/jxnl/instructor/assets/45760326/b56d4714-bea4-4a7d-bb5d-f673f5dc2550

Also added a docs page which shows a new snippet on how to generate these .jsonl files

**Comment by slavakurilyak** [comment]
When can we expect the next release to include this? 

Excited to use this batch CLI command

**Comment by ivanleomk** [comment]
@slavakurilyak we just released v1.3.4 which has it!

**Pull request merged**